app-backend/db/src/main/scala/ProjectDao.scala

```diff
@@ -259,28 +262,52 @@ object ProjectDao
       WHERE
       scenes.id NOT IN (
        SELECT scene_id
-       FROM scenes_to_projects
-       WHERE project_id = ${projectId} AND accepted = true
-      )
-      AND """ ++ inClause
+       FROM scenes_to_layers
+       WHERE project_layer_id = ${projectLayerId} AND accepted = true
+      ) AND """ ++ Fragments.in(fr"scenes.id", sceneIds)
+
+  def getProjectLayerId(projectLayerIdO: Option[UUID], project: Project): UUID =
+    (projectLayerIdO, project.defaultLayerId) match {
+      case (Some(projectLayerId), _) => projectLayerId
+      case (_, Some(defaultLayerId)) => defaultLayerId
+      case _ =>
+        throw new Exception(s"Project ${project.id} does not have any layer")
+    }
+
+  def addScenesToProject(sceneIds: NonEmptyList[UUID],
+                         projectId: UUID,
+                         isAccepted: Boolean,
+                         projectLayerIdO: Option[UUID]): ConnectionIO[Int] = {
     for {
       project <- ProjectDao.unsafeGetProjectById(projectId)
       user <- UserDao.unsafeGetUserById(project.owner)
-      sceneQueryResult <- sceneIdWithDatasourceF
+      projectLayerId = getProjectLayerId(projectLayerIdO, project)
+      sceneIdWithDatasource <- sceneIdWithDatasourceF(sceneIds, projectLayerId)
         .query[(UUID, Datasource)]
         .to[List]
-      sceneToProjectInserts <- {
-        val scenesToProject: List[SceneToProject] = sceneQueryResult.map {
+      sceneToLayerInserts <- {
+        val scenesToLayer: List[SceneToLayer] = sceneIdWithDatasource.map {
+          case (sceneId, datasource) =>
+            createScenesToLayer(sceneId, projectLayerId, datasource, isAccepted)
+        }
+        val insertScenesToLayers =
+          "INSERT INTO scenes_to_layers (scene_id, project_layer_id, accepted, scene_order, mosaic_definition) VALUES (?, ?, ?, ?, ?)"
+        Update[SceneToLayer](insertScenesToLayers).updateMany(scenesToLayer)
+      }
+      // TODO: delete this when we can get rid of scenes_to_projects entirely
```

Comment:
Can you add an issue for this TODO - maybe start it as a running list of things we can get rid of after we move everything to layers

Reply:
Will do.